### DISCUSSION TITLE
Join using three tables and aggregate fails

### DISCUSSION THREAD

**Discussion by femto-code** [ujjwalguptaofficial/JsStore#291] [Q&A] [answered]
Hi, I need help again with a join using three tables. I have tried the following:

```
var results = await connection.select({
    from: "transactions",
    where: {
      date: {
        '>': start,
        '<': end
      }
    },
    join:[{
        with:"labels",
        on: "transactions.label=labels.id",
        as: {
            id: "labelsId",
        }
    },{
        with:"categories",
        on: "labels.category=categories.id",
        as: {
            id: "catId",
        }
    }],
    groupBy: 'category',      // OR: 'catId'
    aggregate: {
      sum: "amount"
    }
});
console.log(results);
```
I would like to group the transactions by their category (which is defined in _labels.category_, equals/joined with _categories.id_) and get the overall sum of **amount**s (column in _transactions_) per category.
However - no matter what I set for the aggregate, the result is always wrong, e.g. the **sum(amount)** column always has the same value as the **amount** column, so is not the corrected sum but just the value of one entry (in transactions).
Thanks for your help!

Edit: I already tried _tablename.columnname_ scheme but that also did not work.

**Comment by ujjwalguptaofficial** [comment]
Seems like there is some bug when using aggregate with join, i am taking a look.

**Reply by femto-code** [reply]
That would be very helpful, thank you!

**Reply by femto-code** [reply]
The weird thing is that I tried an example with join in IDB studio and it worked there.
Can't figure out why it is not working in my case. Because of _double_ join?

**Reply by ujjwalguptaofficial** [reply]
no, the issue was even with the single join when using aggregate and groupby.

**Comment by ujjwalguptaofficial** [comment, marked as answer]
@femto-code fixed in v 4.4.4. Please check and let me know.

So it was because - the aggregation was runing two times causing everything to be 1. Please check and let me know.

**Reply by femto-code** [reply]
Confirming that it is working now.
